subprojects/packagefiles/iir/iir/meson.build

```diff
@@ -0,0 +1,32 @@
+sources = files(
+  'Biquad.cpp',
+  'Butterworth.cpp',
+  'Cascade.cpp',
+  'ChebyshevI.cpp',
+  'ChebyshevII.cpp',
+  'Custom.cpp',
+  'PoleFilter.cpp',
+  'RBJ.cpp')
+
+# Maybe disable exceptions
+if not get_option('exceptions')
+  add_project_arguments('-DIIR1_NO_EXCEPTIONS', 'cpp')
+endif
+
+lib = library('iir', 
+              sources,
+              include_directories : incdirs,
+              install: true)
+
+libiir_dep = declare_dependency(include_directories : incdirs,
+                                link_with: lib)
+
+pkgconfig = import('pkgconfig')
+pkgconfig.generate(lib,
+                   description: 'IIR is an an infinite impulse response (IIR) filter library',
+                   version: meson.project_version())
+
+cmake = import('cmake')
+cmake.write_basic_package_version_file(name: meson.project_name(),
```

Comment:
As I said buried in another comment, this version file isn't very useful on its own, because unlike pkg-config, you need several files at once for a full cmake dependency.

Also, note that this write_basic_package_version_file() method will cause building to fail with `ERROR: Unable to find cmake` if cmake is not installed, which is a bit unfortunate.

Reply:
I'll delete the cmake config creation chunk.

You mentioned `pkgconfig` still needing a version kwarg, implying the above `version: meson.project_version()` is not the way to do it.

Because this is a cmake package, it doesn't have a `configure.ac` to lift out the prior `soversion` details like you helped me with in the speexdsp.

What's the easiest way to satisfy this?

Reply:
Removed cmake block.

Reply:
No :) I said `library()` needs a version kwarg, to ensure that it gets created with the correct soname.

The upstream cmake configuration is very simple though, it sets cmake's `SOVERSION 1 VERSION ${PROJECT_VERSION}`

project version is 1.9.2 of course, so that maps to `version: meson.project_version()` in Meson, no need to set the soversion separately as Meson defaults to "the first part of version" if not set.

Reply:
> project version is 1.9.2 of course, so that maps to version: meson.project_version() in Meson

Thanks; now set:

``` python
lib = library('iir',
              sources,
              include_directories : incdirs,
              install: true,
              version: meson.project_version())
```